pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -285,10 +285,11 @@ func getLatestConfig(url string, iteration int,
 	getconfigCtx *getconfigContext) bool {
 
 	log.Tracef("getLatestConfig(%s, %d)", url, iteration)
-	// If we haven't yet published our certificates we defer to ensure
-	// that the controller has our certs and can add encrypted secrets to
-	// our config.
-	if !getconfigCtx.zedagentCtx.publishedEdgeNodeCerts {
+	// On first boot, if we haven't yet published our certificates we defer
+	// to ensure that the controller has our certs and can add encrypted
+	// secrets to our config.
+	if getconfigCtx.zedagentCtx.currentBootReason == types.BootReasonFirst &&
```

Comment:
As I can see, `getconfigCtx.zedagentCtx.currentBootReason` never matches `types.BootReasonFirst`. We can see only `BootReasonNone` by default, `BootReasonPoweroffCmd` or `BootReasonRebootCmd` if we receive them during [parsing of config](https://github.com/lf-edge/eve/blob/157e774fe132c90ea353adb0f79cb9d561e487cb/pkg/pillar/cmd/zedagent/parseconfig.go#L2656) or from the [local commands](https://github.com/lf-edge/eve/blob/157e774fe132c90ea353adb0f79cb9d561e487cb/pkg/pillar/cmd/zedagent/localinfo.go#L735).
Disabling of defer here gives me the problem with `restartAttestation` call few lines below. We stuck in putting `EventRestart` into `eventTrigger` channel because we [call](https://github.com/lf-edge/eve/blob/157e774fe132c90ea353adb0f79cb9d561e487cb/pkg/pillar/cmd/zedagent/zedagent.go#L1421) `attestModuleStart` after [waiting](https://github.com/lf-edge/eve/blob/157e774fe132c90ea353adb0f79cb9d561e487cb/pkg/pillar/cmd/zedagent/zedagent.go#L1411) for the first `getLatestConfig` return. Probably it would be great to change the order or to wrap `restartAttestation` into additional check of attestation started.

Reply:
Goot catch! First issue is that "current*" should really be named "requested*" since it is what the controller or microservices has requested to do, and I should we checking bootReason here.

But I don't understand how this can cause issues with restartAttestation. That is only called if we get SenderStatusForbidden. But are you saying we should add a check that restartAttestation doesn't make sense unless we already tried attestation at least once?

Reply:
> But I don't understand how this can cause issues with restartAttestation. That is only called if we get SenderStatusForbidden. But are you saying we should add a check that restartAttestation doesn't make sense unless we already tried attestation at least once?

Correct, logically we should not restart attestation if it not started. Technically we must not call restartAttestation if our handlers are not ready, because we will be blocked by submitting to the non-buffered channel.

Reply:
We've already started the goroutine and done the init so the FSM should be ready to pick up more events. But I'll add a check for this.

Reply:
@giggsoff please review with the added counter for attestion tries.